### PR TITLE
Fix render issue when isInEditMode()

### PR DESCRIPTION
**Pull request by Namolem**
EmojiTextView#setText() uses EmojiManager and get called in constructor

Therefore Android Studio's layout preview fails to render any subclass of EmojiTextView with "Please install an EmojiProvider" exception

http://prntscr.com/s2ofqp

`java.lang.IllegalStateException: Please install an EmojiProvider through the EmojiManager.install() method first.
	at com.vanniktech.emoji.EmojiManager.verifyInstalled(EmojiManager.java:213)
	at com.vanniktech.emoji.EmojiManager.replaceWithImages(EmojiManager.java:170)
	at com.vanniktech.emoji.EmojiTextView.setText(EmojiTextView.java:50)
	at android.widget.TextView.<init>(TextView.java:1616)
	at android.widget.TextView.<init>(TextView.java:968)
	at androidx.appcompat.widget.AppCompatTextView.<init>(AppCompatTextView.java:99)
	at androidx.appcompat.widget.AppCompatTextView.<init>(AppCompatTextView.java:95)
	at com.vanniktech.emoji.EmojiTextView.<init>(EmojiTextView.java:21)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.jetbrains.android.uipreview.ViewLoader.createNewInstance(ViewLoader.java:404)
	at org.jetbrains.android.uipreview.ViewLoader.loadClass(ViewLoader.java:187)
	at org.jetbrains.android.uipreview.ViewLoader.loadView(ViewLoader.java:145)
	at com.android.tools.idea.rendering.LayoutlibCallbackImpl.loadView(LayoutlibCallbackImpl.java:309)
	at android.view.BridgeInflater.loadCustomView(BridgeInflater.java:417)
	at android.view.BridgeInflater.loadCustomView(BridgeInflater.java:428)
	at android.view.BridgeInflater.createViewFromTag(BridgeInflater.java:332)
	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:961)
	at android.view.LayoutInflater.rInflate_Original(LayoutInflater.java:1123)
	at android.view.LayoutInflater_Delegate.rInflate(LayoutInflater_Delegate.java:72)
	at android.view.LayoutInflater.rInflate(LayoutInflater.java:1097)
	at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:1084)
	at android.view.LayoutInflater.rInflate_Original(LayoutInflater.java:1126)
	at android.view.LayoutInflater_Delegate.rInflate(LayoutInflater_Delegate.java:72)
	at android.view.LayoutInflater.rInflate(LayoutInflater.java:1097)
	at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:1084)
	at android.view.LayoutInflater.rInflate_Original(LayoutInflater.java:1126)
	at android.view.LayoutInflater_Delegate.rInflate(LayoutInflater_Delegate.java:72)
	at android.view.LayoutInflater.rInflate(LayoutInflater.java:1097)
	at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:1084)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:682)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:501)
	at android.view.BridgeInflater.inflate(BridgeInflater.java:383)
	at android.view.View.inflate(View.java:25764)
	at kz.btsdigital.aitu.channel.posts.widget.ChannelHeaderView.<init>(ChannelHeaderView.kt:66)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.jetbrains.android.uipreview.ViewLoader.createNewInstance(ViewLoader.java:404)
	at org.jetbrains.android.uipreview.ViewLoader.loadClass(ViewLoader.java:187)
	at org.jetbrains.android.uipreview.ViewLoader.loadView(ViewLoader.java:145)
	at com.android.tools.idea.rendering.LayoutlibCallbackImpl.loadView(LayoutlibCallbackImpl.java:309)
	at android.view.BridgeInflater.loadCustomView(BridgeInflater.java:417)
	at android.view.BridgeInflater.loadCustomView(BridgeInflater.java:428)
	at android.view.BridgeInflater.createViewFromTag(BridgeInflater.java:332)
	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:961)
	at android.view.LayoutInflater.rInflate_Original(LayoutInflater.java:1123)
	at android.view.LayoutInflater_Delegate.rInflate(LayoutInflater_Delegate.java:72)
	at android.view.LayoutInflater.rInflate(LayoutInflater.java:1097)
	at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:1084)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:682)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:501)
	at com.android.layoutlib.bridge.impl.RenderSessionImpl.inflate(RenderSessionImpl.java:328)
	at com.android.layoutlib.bridge.Bridge.createSession(Bridge.java:396)
	at com.android.tools.idea.layoutlib.LayoutLibrary.createSession(LayoutLibrary.java:209)
	at com.android.tools.idea.rendering.RenderTask.createRenderSession(RenderTask.java:608)
	at com.android.tools.idea.rendering.RenderTask.lambda$inflate$6(RenderTask.java:734)
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1590)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
`

### CODE REVIEW

**Comment by mario** [comment]
Should the same be done for edittext etc?

**Comment by Namolem** [comment]
> Should the same be done for edittext etc?

had no problems with that classes yet, can also submit fix for them in this PR if needed

**Comment by Namolem** [comment]
seems not needed for EditText, Emoji's stuff is being used only when 'onTextChanged' called

**Comment by Namolem** [comment]
> Nice, thanks!
> 
> Let's also add that in `EmojiEditText` and `EmojiButton`. Have not tested it, but `EmojiButton` should crash with the same error as it calls the `EmojiManager` in `setText`. `EmojiEditText` does not crash because `onTextChanged` gets not called by AS. For symmetry and because that may change at any time we should add it also there.

done

**Comment by mario** [comment]
Nice. @vanniktech can you make me admin so I can merge stuff like this? :) Thanks!

**Comment by Namolem** [comment]
Could you please also publish 0.6.1 fix this fix?

**Comment by mario** [comment]
@Namolem we will soon (in a week? as soon as I finish one more thing) have 0.7.0.

**Comment by vanniktech** [comment]
I synced my travis account again and I think we should be having updates now again on the PR's

File: emoji/src/main/java/com/vanniktech/emoji/EmojiEditText.java [hunk 1]
```diff
@@ -48,6 +48,8 @@ public EmojiEditText(final Context context, final AttributeSet attrs) {
   }
 
   @Override @CallSuper protected void onTextChanged(final CharSequence text, final int start, final int lengthBefore, final int lengthAfter) {
+    if (isInEditMode()) return;
```
Comment (vanniktech) [inline emoji/src/main/java/com/vanniktech/emoji/EmojiEditText.java:51]:
```suggestion
    if (isInEditMode()) {
      return;
    }
```

**Comment by vanniktech** [comment]
Thanks!

**Comment by Namolem** [comment]
@vanniktech could you please release this PR. We we're really looking forward to it

**Pull request merged**